### ISSUE TITLE
 Compatibility with Gooee and Road Builder mod

### ISSUE THREAD

**Issue by Schnubbels** [slyh/Cities2-TrafficLightsEnhancement#107]
With the release of the new Road Builder Mod, TLE is unfortunately no longer compatible with the new mod. Both mods cannot be used at the same time. Is it possible to ensure compatibility of both mods with an update?

**Comment by zeroibis** [comment]
I have not had any issues when testing the pre-release version or post pubic release road builder mod and Traffic Lights Enhancement.

I have been running fine and able to execute all operations with the following mods:
75724: ExtraLib 1.0.0.0
75426: I18n EveryWhere 1.2.2.0
74417: Unified Icon Library 1.0.9.0
80529: ExtraAssetsImporter 1.0.0.0
78903: Asset Packs Manager 1.4.2.0
75804: Skyve [BETA] 0.10.2.2
87190: Road Builder [BETA] 0.1.0.0
85217: [G87] Color Surfaces 1.0.0.0
86605: All Aboard! (Faster Boarding Mod) 1.0.0.0
84571: Bye Bye Homeless 1.0.0.0
80754: Tram Variation Asset Pack 1.0.5.0
75190: School Capacity Balancer 1.0.0.0
76546: NoDogs 1.0.0.0
74324: Move It 0.4.3.37329
79237: First Person Camera Continued 1.0.0.0
78601: No Pollution 1.0.0.0
75745: Achievement Enabler 1.0.0.0
80095: Traffic 0.2.0.0
78960: Traffic Lights Enhancement Alpha 0.0.14.0
75816: Line Tool 0.9.6.0
75250: Better Bulldozer 1.3.7.0
74604: Anarchy 1.6.4.0

I do have the road builder mod set to load first in paradox mods although it is second now that I installed Skyve.

**Comment by zeroibis** [comment]
My understanding is that if people still have Gooee installed then it creates an issue with road builder and TLE. 

For users who have Gooee the solution is to uninstall Goee.

**Comment by vinceingithub** [comment]
> My understanding is that if people still have Gooee installed then it creates an issue with road builder and TLE.
> 
> For users who have Gooee the solution is to uninstall Goee.

What about our mods which need Gooee?

**Comment by Schnubbels** [comment]
Gooe was the problem. After deleting the mod everything works.

**Comment by vinceingithub** [comment]
> Gooe was the problem. After deleting the mod everything works. But it worked also, if only Road Builder **or** TLE were active.

Is it possible to implement Gooee compatibibility?

**Comment by gaborposz** [comment]
I tried to disable Gooee, but then I have no UI for any mods, i.e. none of them is usable. :(

![image](https://github.com/user-attachments/assets/1585307f-04e1-4f80-bdfe-abe123edad54)

**Comment by slyh** [comment]
I can't reproduce this issue with only TLE, Traffic and Road Builder installed. However, I have implemented a fix in the canary version based on the error message received. If anyone still encounters issues after deleting all files related to Gooee, please try the canary version.

For instructions on removing Gooee completely, please refer to issue #105.

> Is it possible to implement Gooee compatibility?

Are there any mods that still require Gooee?

**Comment by Bendak** [comment]
> > I can't reproduce this issue with only TLE, Traffic and Road Builder installed. However, I have implemented a fix in the canary version based on the error message received. If anyone still encounters issues after deleting all files related to Gooee, please try the canary version.
> > For instructions on removing Gooee completely, please refer to issue #105.
> > > Is it possible to implement Gooee compatibility?
> > 
> > 
> > Are there any mods that still require Gooee?
> > > I tried to disable/remove Gooee, but then I have no UI for any mods, i.e. none of them is usable. :(
> > > ![image](https://private-user-images.githubusercontent.com/47426565/358503075-1585307f-04e1-4f80-bdfe-abe123edad54.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM4MjUzMzIsIm5iZiI6MTcyMzgyNTAzMiwicGF0aCI6Ii80NzQyNjU2NS8zNTg1MDMwNzUtMTU4NTMwN2YtMDRlMS00ZjgwLWJkZmUtYWJlMTIzZWRhZDU0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE2VDE2MTcxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJjZjNiNWE0OWJiOWRmZDcyMTAzMmJmZDIxMWNkYzU2YmYwYjI0MjUxMWJiMDM1ZTI5OGZhYmZmMTAwZjE1YWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.MIlwoIdNY8rc_XsK4kfDXS-ZeY1hxF0DPRUiA0Vthfg)
> > 
> > 
> > Please manually delete any files related to Gooee, as unsubscribing alone will not remove residual files.
> 
> Tried to remove Gooe and had similar issue. all mods, and all UI (e.g TLE UI) did not show up. I tested with it disabled and removed. It basically renders mods with custom UI unusable.
> 
> Will try to manually remove it, but if it is disabled, it doesn't make sense for it to cause other mods ui to crash.

Yes, i did remove the settings folder and now it works as well. Not sure how a disabled mod would impact other mods, but it does.

**Comment by vinceingithub** [comment]
> > Is it possible to implement Gooee compatibility?
> 
> Are there any mods that still require Gooee?

Yes. Forest Brush, Legacy Flavor, Traffic, and Zoning Toolkit all require GOOEE.

**Comment by gaborposz** [comment]
Ok, thank you @slyh and @Bendak for the hints, finally I found the solution for the problem:

To fix the compatibility issue it was not enough to disable or remove the **Gooee** mod and delete the `C:\Users\<your_username>\LocalLow\Colossal Order\Cities Skylines II\.cache\Mods\mods_subscribed\` folder, because there were also some additional orphaned Gooee folders on my machine:
```
C:\Users\<your_username>\LocalLow\Colossal Order\Cities Skylines II\Mods\Gooee
C:\Users\<your_username>\LocalLow\Colossal Order\Cities Skylines II\ModsData\Gooee
```
I have deleted these, and now the UI is back, and the compatibility issue between Road Builder and TrafficLightsEnhancemenet is also gone.

**Comment by slyh** [comment]
The issue has been resolved in version 0.2.26.

TLE is now working fine with Road Builder and Gooee. Feel free to reopen this issue if you continue to experience problems.
